cmds/core/gosh/completer_liner.go

```diff
@@ -0,0 +1,216 @@
+// Copyright 2021 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build !tinygo && !plan9 && !goshsmall && goshliner
+// +build !tinygo,!plan9,!goshsmall,goshliner
+
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/peterh/liner"
+	"mvdan.cc/sh/v3/interp"
+	"mvdan.cc/sh/v3/syntax"
+)
+
+// HistFile is the history file.
+// This might, possibly, use GetPid to avoid gosh'es writing over each other
+var HistFile = filepath.Join(os.TempDir(), "gosh.history")
+
+var completion = flag.Bool("comp", false, "Enable tabcompletion and a more feature rich editline implementation")
+
+func lastWord(parser *syntax.Parser, line string) (int, int, string) {
```

Comment:
@mvdan what do you think of this autocomplete tokenization?

Reply:
I see you pushed a new version :) will leave some thoughts in the new patch